pallets/dkg-mmr/src/lib.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (C) 2020-2021 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![warn(missing_docs)]
+
+//! A DKG+MMR pallet combo.
+//!
+//! While both DKG and Merkle Mountain Range (MMR) can be used separately,
+//! these tools were designed to work together in unison.
+//!
+//! The pallet provides a standardized MMR Leaf format that is can be used
+//! to bridge DKG+MMR-based networks (both standalone and polkadot-like).
+//!
+//! The MMR leaf contains:
+//! 1. Block number and parent block hash.
+//! 2. Merkle Tree Root Hash of next DKG validator set.
+//! 3. Merkle Tree Root Hash of current parachain heads state.
+//!
+//! and thanks to versioning can be easily updated in the future.
+
+use sp_runtime::traits::{Convert, Hash};
+use sp_std::prelude::*;
+
+use dkg_runtime_primitives::mmr::{DkgNextAuthoritySet, MmrLeaf, MmrLeafVersion};
+use pallet_mmr::primitives::LeafDataProvider;
+
+use codec::Encode;
+use frame_support::traits::Get;
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+#[cfg(test)]
+mod tests;
+
+/// A DKG consensus digest item with MMR root hash.
+pub struct DepositDkgDigest<T>(sp_std::marker::PhantomData<T>);
+
+impl<T> pallet_mmr::primitives::OnNewRoot<dkg_runtime_primitives::MmrRootHash>
+	for DepositDkgDigest<T>
+where
+	T: pallet_mmr::Config<Hash = dkg_runtime_primitives::MmrRootHash>,
+	T: pallet_dkg_metadata::Config,
+{
+	fn on_new_root(root: &<T as pallet_mmr::Config>::Hash) {
+		let digest = sp_runtime::generic::DigestItem::Consensus(
+			dkg_runtime_primitives::DKG_ENGINE_ID,
+			codec::Encode::encode(&dkg_runtime_primitives::ConsensusLog::<
+				<T as pallet_dkg_metadata::Config>::DKGId,
+			>::MmrRoot(*root)),
+		);
+		<frame_system::Pallet<T>>::deposit_log(digest);
+	}
+}
+
+/// Convert DKG secp256k1 public keys into Ethereum addresses
+pub struct DkgEcdsaToEthereum;
```

Comment:
```suggestion
pub struct DKGEcdsaToEthereum;
```
and update this everywhere `Dkg` to `DKG`